components/board/customlinux/board.go

```diff
@@ -0,0 +1,131 @@
+//go:build linux
+
+// Package customlinux implements a board running Linux.
+// This is an Experimental package
+package customlinux
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"periph.io/x/host/v3"
+
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/board/genericlinux"
+	"go.viam.com/rdk/resource"
+)
+
+const modelName = "customlinux"
+
+func init() {
+	if _, err := host.Init(); err != nil {
+		golog.Global().Debugw("error initializing host", "error", err)
+	}
+
+	resource.RegisterComponent(
+		board.API,
+		resource.DefaultModelFamily.WithModel(modelName),
+		resource.Registration[board.Board, *Config]{
+			Constructor: createNewBoard,
+		})
+}
+
+// customLinuxBoard wraps the genericlinux board type so that both can implement their own Reconfigure function.
+type customLinuxBoard struct {
+	*genericlinux.SysfsBoard
+	mu     sync.Mutex
+	logger golog.Logger
+}
+
+func createNewBoard(
+	ctx context.Context,
+	_ resource.Dependencies,
+	conf resource.Config,
+	logger golog.Logger,
+) (board.Board, error) {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return nil, err
+	}
+
+	pinDefs, err := parseBoardConfig(newConf.PinConfigFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	gpioMappings, err := genericlinux.GetGPIOBoardMappingFromPinDefs(pinDefs)
+	if err != nil {
+		return nil, err
+	}
+
+	boardConfig := createGenericLinuxConfig(newConf)
+	sysfsB, err := genericlinux.NewSysfsBoard(ctx, conf.ResourceName().AsNamed(), &boardConfig, gpioMappings, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	gb, ok := sysfsB.(*genericlinux.SysfsBoard)
+	if !ok {
+		return nil, errors.New("error creating board object")
+	}
+
+	b := customLinuxBoard{SysfsBoard: gb, logger: logger}
```

Comment:
From the go sdk, call .Name() on the customlinux board and make sure we don't get a nil pointer error.

Reply:
tested, no nil pointer errors